File: utilCode/src/main/java/cn/rubintry/utilcode/logger/ILogger.java
package cn.rubintry.utilcode.logger;

/**
 * @author logcat
 */
public interface ILogger {
    void debug(String message);

    void debug(String message , Throwable ex);

    void error(String message);

    void error(String message , Throwable ex);


    void warn(String message);

    void warn(String message , Throwable ex);
}

File: utilCode/src/main/java/cn/rubintry/utilcode/logger/ILogger.kt
package cn.rubintry.utilcode.logger

/**
 * @author logcat
 */
interface ILogger {
    open fun debug(message: String?)
    open fun debug(message: String?, ex: Throwable?)
    open fun error(message: String?)
    open fun error(message: String?, ex: Throwable?)
    open fun warn(message: String?)
    open fun warn(message: String?, ex: Throwable?)
}